src/Orchard.Web/Modules/Orchard.OpenId/Module.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Logging;
+using Orchard.Environment.Shell;
+using Orchard.FileSystem.AppData;
+using Orchard.OpenId.Indexes;
+using Orchard.OpenId.Models;
+using Orchard.OpenId.Services;
+using Orchard.Security;
+using Orchard.Settings;
+using Orchard.Users.Models;
+using System;
+
+namespace Orchard.OpenId
+{
+    public class Startup : StartupBase
+    {     
+        private readonly string _certificateFullPath;
+        private const string certificateFileName = "Certificate.pfx";
+        private readonly string tenantUrlPrefix;
+        private readonly string tenantName;
+        public Startup(ShellSettings shellSettings, IAppDataFolder appDataFolder, ILoggerFactory loggerFactory)   
+        {
+            tenantName = shellSettings.Name;
+            tenantUrlPrefix = string.IsNullOrWhiteSpace(shellSettings.RequestUrlPrefix) ? "" : "/" + shellSettings.RequestUrlPrefix;
+            _certificateFullPath = appDataFolder.Combine(shellSettings.Name, certificateFileName);
+        }
+
+        public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var baseUrl = serviceProvider.GetRequiredService<ISiteService>().GetSiteSettingsAsync().Result.BaseUrl;
+            baseUrl = baseUrl.EndsWith("/") ? baseUrl.Remove(baseUrl.Length - 1) : baseUrl;
+            var tenantUrl = baseUrl + tenantUrlPrefix;
+
+            builder.UseOpenIddict();
+            builder.UseJwtBearerAuthentication(new JwtBearerOptions()
+            {
+                AutomaticAuthenticate = true,
+                AutomaticChallenge = true,
+#if DEBUG
+                RequireHttpsMetadata = false,
+#endif
+                Audience = tenantUrl,
+                Authority = tenantUrl
+            });
+        }
+
+        public override void ConfigureServices(IServiceCollection serviceCollection)
+        {
+            serviceCollection.AddScoped<OpenIdApplicationIndexProvider>();
+            serviceCollection.AddScoped<OpenIdTokenIndexProvider>();            
+            serviceCollection.TryAddScoped<IOpenIdApplicationManager, OpenIdApplicationManager>();
+            var openIddictBuilder = serviceCollection.AddOpenIddict<User, Role, OpenIdApplication, OpenIdAuthorization, OpenIdScope, OpenIdToken>()
+            .AddApplicationStore<OpenIdApplicationStore>()
+            .AddTokenStore<OpenIdTokenStore>()
+            .AddUserStore<OpenIdUserStore>()
+            .AddUserManager<OpenIdUserManager>()
+            //.Configure(options => options.DataProtectionProvider = DataProtectionProvider.Create(tenantName))
```

Comment:
@PinpointTownes The code in this commented line is what we should do to support multitenant configuration if we were using standard tokens? Is it ok? Or there is something else to do?. Can you share any sample code where I can see what is needed to support multitenant configuration using JWT token, I don't have clear that point.